geth/jail/internal/loop/loop.go

```diff
@@ -65,19 +103,23 @@ func (l *Loop) VM() *vm.VM {
 // Add puts a task into the loop. This signals to the loop that this task is
 // doing something outside of the JavaScript environment, and that at some
 // point, it will become ready for finalising.
-func (l *Loop) Add(t Task) {
+func (l *Loop) Add(t Task) error {
 	l.lock.Lock()
+	defer l.lock.Unlock()
+	if !l.isAccepting() {
+		return errors.New("The loop is closed and no longer accepting tasks")
+	}
 	t.SetID(atomic.AddInt64(&l.id, 1))
 	l.tasks[t.GetID()] = t
-	l.lock.Unlock()
+	return nil
 }
 
 // Remove takes a task out of the loop. This should not be called if a task
 // has already become ready for finalising. Warranty void if constraint is
 // broken.
 func (l *Loop) Remove(t Task) {
 	l.remove(t)
-	go l.Ready(nil)
+	go l.Ready(nil) // nolint: errcheck
```

Comment:
*Out of scope for this PR, but once have to ask the original authors.* 

The `nil` is only sent to give the loop in `Run()` the chance to check, if the context contains an error. It's not very elegant to give the `nil` this special semantic.